stac_generator/raster_data/generator.py

```diff
@@ -29,105 +39,133 @@ def __init__(
-        if 'epsg' in source_df.columns:
-            source_df['epsg'] = source_df['epsg'].astype(int)
+
+        # Ensure the EPSG column is properly formatted
+        if "epsg" in source_df.columns:
+            source_df["epsg"] = source_df["epsg"].astype(int)
+
+        # Parse the 'bands' column into a list if it is a JSON-encoded string
+        if "bands" in source_df.columns:
+
+            def parse_bands(bands: Any) -> Any:
+                if isinstance(bands, str):
+                    try:
+                        return json.loads(bands)
+                    except json.JSONDecodeError as e:
+                        raise ValueError(f"Error parsing 'bands': {bands}") from e
+                return bands
+
+            source_df["bands"] = source_df["bands"].apply(parse_bands)
+
+        # Pass the modified DataFrame to the base class
         super().__init__(
             source_df=source_df,
             collection_cfg=collection_cfg,
             catalog_cfg=catalog_cfg,
             href=href,
         )
 
-    def create_item_from_config(self, source_cfg: RasterSourceConfig) -> List[pystac.Item]:
-        logging.info(f"Processing raster from: {source_cfg.location}")
-        
+    def create_item_from_config(self, source_cfg: RasterSourceConfig) -> list[pystac.Item]:
+        logging.info("Processing raster from: %s", source_cfg.location)
+
         try:
             with rasterio.open(source_cfg.location) as src:
                 bounds = src.bounds
                 bbox = [bounds.left, bounds.bottom, bounds.right, bounds.top]
                 transform = src.transform
                 crs = src.crs
-                
-                geometry = {
-                    "type": "Polygon",
-                    "coordinates": [[
-                        [bounds.left, bounds.bottom],
-                        [bounds.left, bounds.top],
-                        [bounds.right, bounds.top],
-                        [bounds.right, bounds.bottom],
-                        [bounds.left, bounds.bottom]
-                    ]]
-                }
-
-                # Create base item
+                shape = [src.height, src.width]
+
+                # Create geometry as Shapely Polygon
+                geometry = Polygon(
+                    [
+                        (bounds.left, bounds.bottom),
+                        (bounds.left, bounds.top),
+                        (bounds.right, bounds.top),
+                        (bounds.right, bounds.bottom),
+                        (bounds.left, bounds.bottom),
+                    ]
+                )
+                geometry_geojson = mapping(geometry)
+
+                if source_cfg.datetime is None:
+                    raise ValueError("source_cfg.datetime cannot be None")
+
+                # Ensure datetime is timezone-aware
+                if source_cfg.datetime.tzinfo is None:
+                    datetime_aware = source_cfg.datetime.replace(tzinfo=UTC)
```

Comment:
I think it will be problematic to assume UTC. There is probably not going to be any timezone information in the source_cfg. Timezone is a function of location though and we do have that information. Im not sure of the best way to do this but there are almost certainly packages that give you the timezone from a lat/lon or other location input.

Reply:
So we can get the location from geotiff file. Use a point or centroid.

Reply:
done